woocommerce/woocommerce
# Use correct option type for password generation option

### Changes proposed in this Pull Request:

This PR fixes an issue in which we would mistype a Woo option, that's a string of "yes" "no" assuming it's boolean, and would up always generating a password. I also fixed another instance of that, hoping it would fix this PR https://github.com/woocommerce/woocommerce/issues/53117 but it's not the cause.

Closes https://github.com/woocommerce/woocommerce/issues/53276



### How to test the changes in this Pull Request:



Using the [WooCommerce Testing Instructions Guide](https://github.com/woocommerce/woocommerce/wiki/Writing-high-quality-testing-instructions), include your detailed testing instructions:

1. In WooCommerce -> Settings -> Account and Privacy: Enable account creation after Checkout and don't send a password
<img width="872" alt="image" src="https://github.com/user-attachments/assets/d9838d0c-e7d9-492d-8de6-32b6717a43a6">

2. Logged out, go to Checkout, place an order without creating a checkout.
3. On post checkout, create an account and note the password.
4. Log out.
5. Try logging in again with the same account. It should pass



### Changelog entry












<details>

<summary>Changelog Entry Details</summary>

#### Significance




#### Type




#### Message 

Correctly respect the generate password option for accounts created post Checkout.

</details>

<details>

<summary>Changelog Entry Comment</summary>

#### Comment 

</details>
